gateway/handle.go

```diff
@@ -42,6 +45,11 @@ import (
 	"github.com/rudderlabs/rudder-server/utils/types"
 )
 
+var (
+	streamMsgValidator = stream.NewMessageValidator()
```

Comment:
We can probably avoid this global and pass it as a dependency?
```
type Handle struct {
	config           *config.Config
	logger           logger.Logger
	stats            stats.Stats
	tracer           stats.Tracer
	messageValidator func(msg *stream.Message) error
        ...

}
```

And the setup will look something like this:
```
func (gw *Handle) Setup(
	ctx context.Context,
	config *config.Config, logger logger.Logger, stat stats.Stats,
	application app.App, backendConfig backendconfig.BackendConfig, jobsDB, errDB jobsdb.JobsDB,
	rateLimiter throttler.Throttler, versionHandler func(w http.ResponseWriter, r *http.Request),
	rsourcesService rsources.JobService, transformerFeaturesService transformer.FeaturesService, sourcehandle sourcedebugger.SourceDebugger,
	opts ...OptFunc,
) error {
	gw.config = config
	gw.logger = logger
	gw.stats = stat\
	gw.messageValidator = stream.NewMessageValidator()
```

And usage will look something like:
```
	  for _, msg := range messages {
		if err := gw.messageValidator(&msg); err != nil {
			return nil, errors.New("invalid message")
		}
                 ...
           }
```

Reply:
@lvrach I forgot, why we decided to declare it globally instead of injecting it as a dependency?